src/org/openj9/envInfo/MachineInfo.java

```diff
@@ -201,10 +203,32 @@ private void getSysInfo() {
 		putInfo(new Info("sysArch", SYS_ARCH_CMD, ce.execute(SYS_ARCH_CMD), null));
 		putInfo(new Info("procArch", PROC_ARCH_CMD, ce.execute(PROC_ARCH_CMD), null));
 		String microArchOutput = ce.execute(MICRO_ARCH_CMD);
+		// Check for specific machine versions and set the microArch accordingly
 		if (microArchOutput.toLowerCase().contains("skylake")) {
-			String microArch = "skylake";
-			putInfo(new Info("microArch", MICRO_ARCH_CMD, microArch, null));
+			microArchOutput = "skylake";
+		} else if (microArchOutput.contains("8562")) {
+			microArchOutput = "IBM z15 T02";
+		} else if (microArchOutput.contains("8561")) {
+			microArchOutput = "IBM z15 T01";
+		} else if (microArchOutput.contains("3907")) {
+			microArchOutput = "IBM z14 ZR1";
+		} else if (microArchOutput.contains("3906")) {
+			microArchOutput = "IBM z14";
+		} else if (microArchOutput.contains("2965")) {
+			microArchOutput = "IBM z13s";
+		} else if (microArchOutput.contains("2964")) {
+			microArchOutput= "IBM z13";
+		} 
+		// Check if the detected microarchitecture is supported
+		Set<String> supportedMicroArchitectures = new HashSet<>(Arrays.asList("skylake", "z15", "z14", "z13"));
+
+		if (supportedMicroArchitectures.stream().anyMatch(microArchOutput.toLowerCase()::contains)) {
+			putInfo(new Info("microArch", MICRO_ARCH_CMD, microArchOutput, null));
+		} else {
+			// The detected microarchitecture is not supported
+			System.out.println("Detected microarchitecture is not supported: " + microArchOutput);
```

Comment:
This message `Detected microarchitecture is not supported` is misleading. We support more than just `"skylake", "z15", "z14", "z13"`. It also means that we have to maintain the `supportedMicroArchitectures` list separately. 

Please remove this block of code. Keep original code and add else to printout microArchOutput value.

Reply:
Should I remove the code in line 229 and print out microArchOutput value there instead?